src/libraries/System.Security.Cryptography/tests/KmacTestDriver.cs

```diff
@@ -0,0 +1,1024 @@
+                    customizationString));
+
+            Assert.Throws<PlatformNotSupportedException>(
+                () => TKmacTrait.HashDataAsync(
+                    MinimalKey,
+                    Stream.Null,
+                    outputLength: 1,
+                    customizationString,
+                    default(CancellationToken)));
+
+            Assert.Throws<PlatformNotSupportedException>(
+                () => TKmacTrait.HashDataAsync(
+                    new ReadOnlyMemory<byte>(MinimalKey),
+                    Stream.Null,
+                    outputLength: 1,
+                    new ReadOnlyMemory<byte>(customizationString),
+                    default(CancellationToken)));
+
+            Assert.Throws<PlatformNotSupportedException>(
+                () => TKmacTrait.HashDataAsync(
+                    new ReadOnlyMemory<byte>(MinimalKey),
+                    Stream.Null,
+                    new Memory<byte>(destination),
+                    new ReadOnlyMemory<byte>(customizationString),
+                    default(CancellationToken)));
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public void CryptographicException_Allocated_KeySize()
+        {
+            if (PlatformKeySizeRequirements?.MinSize - 1 is int smallKey and > 0)
+            {
+                Assert.ThrowsAny<CryptographicException>(
+                    () => TKmacTrait.Create(new byte[smallKey], Array.Empty<byte>()));
+
+                Assert.ThrowsAny<CryptographicException>(
+                    () => TKmacTrait.Create(new ReadOnlySpan<byte>(new byte[smallKey]), default(ReadOnlySpan<byte>)));
+            }
+
+            if (PlatformKeySizeRequirements?.MaxSize + 1 is int largeKey)
+            {
+                Assert.ThrowsAny<CryptographicException>(
+                    () => TKmacTrait.Create(new byte[largeKey], Array.Empty<byte>()));
+
+                Assert.ThrowsAny<CryptographicException>(
+                    () => TKmacTrait.Create(new ReadOnlySpan<byte>(new byte[largeKey]), default(ReadOnlySpan<byte>)));
+            }
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public async Task CryptographicException_OneShot_KeySize()
+        {
+            if (PlatformKeySizeRequirements?.MinSize - 1 is int smallKeySize and > 0)
+            {
+                await AssertOneShotsThrowAnyAsync<CryptographicException>(keySize: smallKeySize);
+            }
+
+            if (PlatformKeySizeRequirements?.MaxSize + 1 is int largeKeySize)
+            {
+                await AssertOneShotsThrowAnyAsync<CryptographicException>(keySize: largeKeySize);
+            }
+        }
+
+        [ConditionalFact(nameof(IsSupported))]
+        public void CryptographicException_Instance_CustomizationStringSize()
+        {
+            if (PlatformMaxCustomizationStringSize + 1 is int tooBigCustomizationString)
+            {
+                Assert.ThrowsAny<CryptographicException>(
```

Comment:
I don't think these size limits were documented, either.

Reply:
I agree the `CryptographicException` should be documented but I am unsure of how to document these limits. They are arbitrarily chosen by OpenSSL - and don't apply to Windows.

Reply:
We have some choice stock phrases here:

```
<exception cref="T:System.Security.Cryptography.CryptographicException">All other errors.</exception>

<exception cref="T:System.Security.Cryptography.CryptographicException">An error occurred during a cryptographic operation.</exception>

<!-- something on these lines, not these actual words -->
<exception cref="T:System.Security.Cryptography.CryptographicException">An error occurred in the hashing or signing operation.</exception>  
```